core/modules/installer/installer.authorize.inc

```diff
@@ -228,10 +228,27 @@ function installer_authorize_update_batch_finished($success, $results) {
       'type' => 'error',
     );
   }
-  // Since we're doing an update of existing code, always add a task for
-  // running update.php.
-  $results['tasks'][] = t('Your projects have been downloaded and updated.');
-  $results['tasks'][] = t('<a href="@update">Run database updates</a>', array('@update' => base_path() . 'core/update.php'));
+  $updates_needed = FALSE;
+  include_once(BACKDROP_ROOT . '/core/includes/install.inc');
+  foreach (module_list() as $module) {
+    $updates = backdrop_get_schema_versions($module);
+    if ($updates !== FALSE) {
+      $current_schema = backdrop_get_installed_schema_version($module);
+      $latest_schema = end($updates);
+      if ($current_schema < $latest_schema) {
```

Comment:
Once `core/update.php` was invoked, the `current_schema` was 999 so if a `latest_schema` was returned it would work.  However, before this was invoked, it was showing as 7102.  Now, most update with be in the 1000 range and only initial ones might be 7000+ but other functions have more complex comparisons.

Reply:
The issue with running  `backdrop_get_installed_schema_version($module)` at this point is that most of the `.install` files have not been included by PHP. And of course, all of the hook_update_N() functions live in the `.install` files. So yes, this will always fail to detect needed update hooks.

Reply:
I had seemed to get results fairly consistently from `backdrop_get_installed_schema_version($module)` ; with `paragraphs`, I got what was in the database (i.e. `7102`) until I had invoked `core/update.php` and then the function in `paragraphs.install` kicked in and provided the value of `999`.

However, I never got anything from `backdrop_get_schema_versions($module)`.

I think this will need someone who understands these functions to have a look and think about how to get the information out at this stage.